core/services/health_service.go

```diff
@@ -0,0 +1,61 @@
+package services
+
+import (
+	"context"
+
+	responses "github.com/PTSS-Support/identity-service/api/dtos/responses/health"
+	HealthStatus "github.com/PTSS-Support/identity-service/domain/enums"
+	"github.com/PTSS-Support/identity-service/infrastructure/repositories"
+	"github.com/PTSS-Support/identity-service/infrastructure/util"
+)
+
+type HealthService struct {
+	healthRepo repositories.HealthRepository
+	logger     util.Logger
+}
+
+func NewHealthService(healthRepo repositories.HealthRepository) *HealthService {
+	return &HealthService{
+		healthRepo: healthRepo,
+		logger:     util.NewLogger("HealthService"),
+	}
+}
+
+func (s *HealthService) CheckReadiness(ctx context.Context) (*responses.HealthResponse, error) {
+	keycloakHealth, err := s.healthRepo.CheckHealth(ctx)
+
+	if err != nil {
+		return &responses.HealthResponse{
+			Status: HealthStatus.StatusDown,
+			Checks: []responses.Check{
+				{
+					Name:   "Keycloak health check",
+					Status: HealthStatus.StatusDown,
+					Data: map[string]interface{}{
+						"error": err.Error(),
+					},
+				},
+			},
+		}, nil
+	}
+
+	return &responses.HealthResponse{
+		Status: HealthStatus.StatusUp,
+		Checks: []responses.Check{
+			{
+				Name:   "Keycloak health check",
+				Status: HealthStatus.StatusUp,
+				Data: map[string]interface{}{
+					"response": keycloakHealth,
+				},
+			},
+		},
+	}, nil
+}
+
+func (s *HealthService) CheckLiveness(ctx context.Context) (*responses.HealthResponse, error) {
+	return &responses.HealthResponse{
+		Status: HealthStatus.StatusUp,
+		Checks: []responses.Check{},
+	}, nil
+}
```

Comment:
Online, i see a lot of different opinions of what the health vs liveness should be doing. In general, what i found is that there should be atleast one endpoint to check the connections etc with other services and check if the pod is ready to accept requests (the readiness endpoint)
The other one is for the scenario where applications running for long periods of time eventually transition to broken states, and cannot recover except by being restarted, and checking for that.
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-command

This could thus from what i see be done by just sending a status: up, but others do it by having the /health endpoint check for memory, if secrets are available etc.
I would recommend choosing either; keep the /health endpoint only and sending status up
or choosing ofr expanding the liveness check to look for; if secrets are loaded in, how much memory is available, etc.

Reply:
I made a chat in claude about this as well btw if you'd want to read, but claude itself is also contradicting itself :/

Reply:
Dit is op basis van  smallrye-health, en daar betekend de liveness check of de server tot http calls kan responden, wat nu de functionality is. De readiness check is kijken of alle dependent services ready zijn.